### PR TITLE
EPL1.0 -> EPL2.0 and contribute.md

### PR DESCRIPTION
**Pull request by svenefftinge**
This PR changes the code license from EPL 1.0 to EPL 2.0 and adds a Contribute.md file.

see https://github.com/eclipse/xtext/issues/1176

### CODE REVIEW

**Comment by svenefftinge** [comment]
updated

File: org.eclipse.xtend.core.tests.java8/.settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -64,7 +64,7 @@ org.eclipse.jdt.ui.keywordthis=false
...
...catchblock" name\="catchblock">// ${todo} Auto-generated catch block\n${exception_var}.printStackTrace();</template><template autoinsert\="true" context\="methodbody_context" deleted\="false" description\="Code in created function stubs" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.methodbody" name\="methodbody">// ${todo} Auto-generated function stub\n${body_statement}</template><template autoinsert\="true" context\="constructorbody_context" deleted\="false" description\="Code in created constructor stubs" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.constructorbody" name\="constructorbody">${body_statement}\n// ${todo} Auto-generated constructor stub</template><template autoinsert\="true" context\="getterbody_context" deleted\="false" description\="Code in created getters" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.getterbody" name\="getterbody">return ${field};</template><template autoinsert\="true" context\="setterbody_context" deleted\="false" description\="Code in created setters" enabled\="true" id\="org.eclipse.wst.jsdt.ui.text.codetemplates.setterbody" name\="setterbody">${field} \= ${param};</template></templates>
```
Comment (szarnekow) [inline org.eclipse.xtend.core.tests.java8/.settings/org.eclipse.jdt.ui.prefs]: jdt.ui prefs are outdated

File: org.eclipse.xtend.core/model/RichStrings.genmodel [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <genmodel:GenModel xmi:version="2.0" xmlns:xmi="http://www.omg.org/XMI" xmlns:ecore="http://www.eclipse.org/emf/2002/Ecore"
-    xmlns:genmodel="http://www.eclipse.org/emf/2002/GenModel" copyrightText="Copyright (c) 2011 itemis AG (http://www.itemis.eu) and others.&#xA;All rights reserved. This program and the accompanying materials&#xA;are made available under the terms of the Eclipse Public License v1.0&#xA;which accompanies this distribution, and is available at&#xA;http://www.eclipse.org/legal/epl-v10.html"
+    xmlns:genmodel="http://www.eclipse.org/emf/2002/GenModel" copyrightText="Copyright (c) 2011 itemis AG (http://www.itemis.eu) and others.&#xA;All rights reserved. This program and the accompanying materials&#xA;are made available under the terms of the Eclipse Public License v2.0&#xA;which accompanies this distribution, and is available at&#xA;http://www.eclipse.org/legal/epl-2.0"
```
Comment (szarnekow) [inline org.eclipse.xtend.core/model/RichStrings.genmodel]: genmodel files

File: org.eclipse.xtend.core/plugin.properties [hunk 3]
```diff
@@ -1,9 +1,9 @@
 ###############################################################################
 # Copyright (c) 2012, 2016 itemis AG (http://www.itemis.eu) and others.
 # All rights reserved. This program and the accompanying materials
-# are made available under the terms of the Eclipse Public License v1.0
+# are made available under the terms of the Eclipse Public License v2.0
```
Comment (szarnekow) [inline org.eclipse.xtend.core/plugin.properties]: properties files

**Comment by cdietrich** [comment]
still test fails

**Pull request merged**